include/hve/arch/intel_x64/base.h

```diff
@@ -93,6 +95,41 @@ class EXPORT_EAPIS_HVE base
     ///
     virtual ~base() = default;
 
+    /// Reason
+    ///
+    /// Example:
+    /// @code
+    /// this->reason();
+    /// @endcode
+    ///
+    /// @expects
+    /// @ensures
+    ///
+    /// @return the basic exit reason handled by this handler
+    ///
+    virtual reason_t reason() = 0;
+
+    /// Cast handler
+    ///
+    /// Example:
+    /// @code
+    /// this->cast_handler();
+    /// @endcode
+    ///
+    /// @expects
+    /// @ensures
+    ///
+    template<typename D> D*
+    cast_handler(gsl::not_null<base *> ptr)
```

Comment:
to be more C++ish, this should probably be `handler_cast`

Reply:
The other function that is missing is, given a reason, and a type, it should return the handler. To do this the base class would need another constructor that takes a list so that it can search it as needed. Obviously, the constructor would take a reference to a list to prevent a copy.